src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/NuGet/ProjectRestoreInfoBuilder.cs

```diff
@@ -46,9 +46,12 @@ internal static IVsProjectRestoreInfo Build(IEnumerable<IProjectVersionedValue<I
                 originalTargetFrameworks = originalTargetFrameworks ??
                     nugetRestoreChanges.After.Properties[NuGetRestore.TargetFrameworksProperty];
 
-                string targetFramework; 
-                if (!update.Value.ProjectConfiguration.Dimensions.TryGetValue(TargetFrameworkProperty, out targetFramework) &&
-                    !nugetRestoreChanges.After.Properties.TryGetValue(TargetFrameworkProperty, out targetFramework))
+                string targetFramework;
+                bool noTargetFramework = 
+                    !update.Value.ProjectConfiguration.Dimensions.TryGetValue(TargetFrameworkProperty, out targetFramework) &&
+                    !nugetRestoreChanges.After.Properties.TryGetValue(TargetFrameworkProperty, out targetFramework);
+
+                if (noTargetFramework || string.IsNullOrWhiteSpace(targetFramework))
```

Comment:
`IsNullOrEmpty` instead of `IsNullOrWhiteSpace`? `targetFramework` should never be set to whitespace. If it does get set then we should know why.